huaweicloud/services/dew/resource_huaweicloud_kms_data_encrypt_decrypt.go

```diff
@@ -0,0 +1,154 @@
+package dew
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API DEW POST /v1.0/{project_id}/kms/encrypt-data
+// @API DEW POST /v1.0/{project_id}/kms/decrypt-data
+func ResourceKmsDataEncryptDecrypt() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKmsDataEncryptDecryptCreate,
+		ReadContext:   resourceKmsDataEncryptDecryptRead,
+		DeleteContext: resourceKmsDataEncryptDecryptDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"encryption_algorithm": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"plain_text": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"cipher_text": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"cipher_data": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"plain_data": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"plain_text_base64": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildDataEncryptDecryptBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"key_id":               utils.ValueIgnoreEmpty(d.Get("key_id")),
+		"encryption_algorithm": utils.ValueIgnoreEmpty(d.Get("encryption_algorithm")),
+		"plain_text":           utils.ValueIgnoreEmpty(d.Get("plain_text")),
+		"cipher_text":          utils.ValueIgnoreEmpty(d.Get("cipher_text")),
+	}
+	return bodyParams
+}
+
+func resourceKmsDataEncryptDecryptCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg                       = meta.(*config.Config)
+		region                    = cfg.GetRegion(d)
+		dataEncryptDecryptHttpUrl = "v1.0/{project_id}/kms/{action}-data"
```

Comment:
Please precheck the value of `action`.
Otherwise, an error will be reported that the URL address does not exist.